src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java

```diff
@@ -61,49 +64,71 @@ public void finishBuildTrace(final BuildData buildData) {
             return;
         }
 
-        final String prefix = BuildPipelineNode.NodeType.PIPELINE.getNormalizedName();
+        final BuildSpanAction buildSpanAction = run.getAction(BuildSpanAction.class);
+        if(buildSpanAction == null) {
+            return;
+        }
+
+        final BuildData pipelineData = buildSpanAction.getBuildData();
+        final String prefix = BuildPipelineNode.NodeType.PIPELINE.getTagName();
         final long endTimeMicros = buildData.getEndTime(0L) * 1000;
         buildSpan.setTag(DDTags.SERVICE_NAME, "jenkins");
         buildSpan.setTag(DDTags.RESOURCE_NAME, buildData.getJobName(null));
         buildSpan.setTag(DDTags.SPAN_TYPE, "ci");
-        buildSpan.setTag(CITags.CI_PROVIDER, "jenkins");
+        buildSpan.setTag(CITags.CI_PROVIDER_NAME, "jenkins");
         buildSpan.setTag(DDTags.LANGUAGE_TAG_KEY, "");
+        buildSpan.setTag(CITags._DD_CI_INTERNAL, false);
         buildSpan.setTag(CITags.USER_NAME, buildData.getUserId());
-        buildSpan.setTag(prefix + CITags._ID, buildData.getBuildId(""));
+        buildSpan.setTag(prefix + CITags._ID, buildData.getBuildTag(""));
         buildSpan.setTag(prefix + CITags._NAME, buildData.getJobName(""));
         buildSpan.setTag(prefix + CITags._NUMBER, buildData.getBuildNumber(""));
-        buildSpan.setTag(prefix + CITags._WORKSPACE, buildData.getWorkspace(""));
-        buildSpan.setTag(CITags.NODE_NAME, buildData.getNodeName(""));
-        buildSpan.setTag(CITags.REPOSITORY_URL, buildData.getGitUrl(""));
-        buildSpan.setTag(CITags.REPOSITORY_BRANCH, buildData.getBranch(""));
-        buildSpan.setTag(CITags.REPOSITORY_COMMIT, buildData.getGitCommit(""));
+        buildSpan.setTag(prefix + CITags._URL, buildData.getBuildUrl(""));
+
+        final String workspace = buildData.getWorkspace("").isEmpty() ? pipelineData.getWorkspace("") : buildData.getWorkspace("");
```

Comment:
Can you explain why `buildData` here might be different than the BuildData instance stored in the Action?

Reply:
Yes, the `BuildData` stored in the Action is updated when the `client.sendPipelineTrace` is executed in the `DatadogGraphListener` listener. 

The `buildData` calculated in the `DatadogBuildListener` will be incomplete if a Jenkins Pipeline is executed. 
E.g., the Git info will be calculated in an inner step of the pipeline, so that info is not available when the root span is created in the `DatadogBuildListener.onStarted` method.

I'll add a comment indicating this fact in the code 👍